src/integration-tests/mem.spec.ts

```
@@ -134,4 +134,21 @@ describe('Memory Test Suite', function() {
 
         verifyMemoryReadResult(mem, 'f1efd4fd7248450c2d13', addrOfArray);
     });
+
+    it('Converts hex to base64 without loss', () => {
+        const base64ToHex = (base64: string): string => Buffer.from(base64, 'base64').toString('hex');
+        const normalize = (original: string): string => original.replace(/^0+/, '').toLowerCase();
+
+        const testCases = [
+            'f',
+            'fe',
+            'fed',
+            'fedc',
+            '0fedc',
```

Just out of curiosity but would gdb send back such weird output (with odd hexadecimal digits?)

Still a good test as it shows the function's resiliency against somewhat malformed data.

---

I don't think GDB will send back anything so weird. And in fact I don't think it will send back anything with an odd number of hex digits, because that would mean there was some strange four-bit  appendix hanging on. On the other hand, I thought the string-->buffer-->string transformation would be smooth, and then I noticed the truncation of odd digits, so I wanted to make sure that the implementation would handle a variety of funny inputs, just in case.

---

Not sure what's best then, if we don't expect funny data but still get some: Fail early or display weird data?